src/connection.rs

```diff
@@ -394,15 +390,14 @@ impl RecvStream {
     }
 
     /// Get the next message sent by the peer over this stream.
-    pub async fn next(&mut self) -> Result<Option<UsrMsgBytes>, RecvError> {
+    pub async fn next(&mut self) -> Result<UsrMsgBytes, RecvError> {
         match self.next_wire_msg().await? {
-            Some(WireMsg::UserMsg(msg)) => Ok(Some(msg)),
-            Some(msg) => Err(RecvError::UnexpectedMsgReceived(msg.to_string())),
-            None => Ok(None),
+            WireMsg::UserMsg(msg) => Ok(msg),
+            msg => Err(RecvError::UnexpectedMsgReceived(msg.to_string())),
         }
     }
 
-    pub(crate) async fn next_wire_msg(&mut self) -> Result<Option<WireMsg>, RecvError> {
```

Comment:
should this be just `wire_msg` or something? As there's only one to get there's no `next`?